### DISCUSSION TITLE
Set password on workbook using JXLS?

### DISCUSSION THREAD

**Discussion by zitotrex** [jxlsteam/jxls#338] [Q&A] [answered]
Hello,

Is it supported or planned to be supported that a password can be set for the workbook as a final pass when JXLS runs? 

And if so, would it work within an excel template using the cell/comment syntax, or would it require writing java code?

**Comment by gastendonk** [comment, marked as answer]
You want to create a password after Jxls processing? That is not planned. Add Apache POI code after your Jxls processing.

**Comment by 313780977** [comment]
String password = "123";
JxlsHelper jxlsHelper = JxlsHelper.getInstance();
jxlsHelper.setUseFastFormulaProcessor(false).setEvaluateFormulas(true).setHideTemplateSheet(true);
Transformer transformer = jxlsHelper.createTransformer(is,os);
PoiTransformer poiTransformer = (PoiTransformer) transformer;

if(!StringUtils.isEmpty(password)){
    Iterator<Sheet> sheetIterator = poiTransformer.getWorkbook().sheetIterator();
    while (sheetIterator.hasNext()){
        Sheet sheet = sheetIterator.next();
        if(null != sheet){
            sheet.protectSheet(password);
        }
    }
}
poiTransformer.getTransformationConfig().getExpressionEvaluator();
jxlsHelper.processTemplate(context, poiTransformer);
